spec/models/access_token_spec.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AccessToken, type: :model do
+  subject { described_class.new token:, scope:, expires_in:, revoked_at:, user: }
+
+  # パディングを除いてトークン文字列として有効な文字を全て含んだ文字列
+  let(:token) { "#{(('a'..'z').to_a + ('A'..'Z').to_a + (1..9).to_a).join}-._~+/" }
+  let(:scope) { 'READ WRITE' }
+  let(:expires_in) { 1.hour }
+  let(:revoked_at) { nil }
+  let(:user) { FactoryBot.create :user }
+
+  context '値が全て適切であるとき' do
+    it { is_expected.to be_valid }
+  end
+
+  describe '.token' do
```

Comment:
以前は気にしていなかったけど、このあたりの describe ってバリデーションの動作確認ですよね。
なので、こういう describe を作って、その下に token, scope, ... とぶら下げるのが良さそう。
```suggestion
  describe '#valid?' do
```

※ テストは invalid? で行っているけど `#valid?` と `#invalid?` は対のものだし、伝わると思うので大目に見てもらおうw

Reply:
改めて今回 spec を読んでて、 `.token` という名前にもやっとしたんですよね。
Ruby で `.token` という書き方をすると、クラスメソッド名のことを指すので ??? となってしまったのです。

Reply:
> Ruby で .token という書き方をすると、クラスメソッド名のことを指すので ??? となってしまったのです。

この辺りの慣習ってどこを見ればわかりますでしょうか。

Reply:
実はちょうど今日、その辺りの話題を調べていました。なので、 `.method` がクラスメソッドを意味するということは知っていたのですが、これまでも `.attribute` の書き方をしていたけど突っ込まれなかったし、修正するにしてもこのPRではないよな、ということで保留にしていました。

https://www.betterspecs.org

属性についてテストする場合はどうやって書くものなのでしょうか。もしかして、そもそも属性に注目してテストは書かなかったりしますか。

Reply:
残念ながらどこで見たのかは覚えてないです。適当にググってみたところだとこのあたりでしょうか。
https://docs.ruby-lang.org/ja/latest/doc/symref.html#period

>これまでも .attribute の書き方をしていたけど突っ込まれなかったし、修正するにしてもこのPRではないよな、ということで保留にしていました。

はい、この PR でなくても良いと思います。
最初のコメントに書いたのですが、以前はあまり違和感を感じなかったので見落としていました。

そういう意味では「突っ込まれなかった」==「問題がひとつもない」ではないので、マージした後にやっぱりこっちのほうが良かった、ということはありえます。
レビューする側も人間なので、すべてのことについて完璧に指摘することは難しいので、そこは容赦してくださいw

>属性についてテストする場合はどうやって書くものなのでしょうか。もしかして、そもそも属性に注目してテストは書かなかったりしますか。

用語を揃えておいたほうが良さそうなので、最初に整理させてください。

* おそらく「属性」と呼んでいるものは「インスタンス変数」と「アクセサメソッド」(仮称)に分かれそうです。
* インスタンス変数というのは @ で始まる変数です (例: @foo, @bar)
* アクセサメソッドというのは attr_reader, attr_writer, attr_accessor で定義するメソッドのことで、インスタンス変数の読み書きをするメソッドを表します
    * 簡単にググった範囲ではこれらの総称を見つけられたので仮称としています

これらを踏まえた上で

* インスタンス変数そのもののテストは行いません。
    * オブジェクトの内部状態で基本的には触ることはできませんし、クラスの利用者は意識しなくてもよいものです。
* アクセサメソッドについてもテストはあまり行いません。
    * validates のところでも議論がありましたが、attr_reader で読み込みメソッドが生えていること、はテストをするメリットが薄いという考えです。

という回答になるかと思います。

一方で、今回コメントしている対象の部分は「インスタンス変数」や「アクセセサメソッド」のテストではないですよね? 提案したように、これらはバリデーションの(バリデーションルールのテストです。
ですから、バリデーションの判定関数である `#valid?` のテストとして記述するのがよいという考えです。
(各属性ごとにバリデーションを行っているので `#valid?` 配下の `token` 属性に関するテストという表現がよさそうだと思っています。

Reply:
> > Ruby で .token という書き方をすると、クラスメソッド名のことを指すので ??? となってしまったのです。
> 
> この辺りの慣習ってどこを見ればわかりますでしょうか。

私も「どこかに『こんな文化があります』が書いてあった」は出せないのだけれど。  
私は「るりま（Ruby リファレンスマニュアル、 https://docs.ruby-lang.org/ja/latest/doc/index.html ）がその書き方をしているから、それに倣っている」だと思っています。

- Class.new
    - https://docs.ruby-lang.org/ja/latest/method/Class/s/new.html
      > singleton method Class.new
- Class#new
    - https://docs.ruby-lang.org/ja/latest/method/Class/i/new.html
      > instance method Class#new

Reply:
> 一方で、今回コメントしている対象の部分は「インスタンス変数」や「アクセセサメソッド」のテストではないですよね? 提案したように、これらはバリデーションの(バリデーションルールのテストです。
> ですから、バリデーションの判定関数である #valid? のテストとして記述するのがよいという考えです。
> (各属性ごとにバリデーションを行っているので #valid? 配下の token 属性に関するテストという表現がよさそうだと思っています。

あー、確かにそうですね。属性ごとにバリデーションを考えていたので、つい属性という観点で見ていましたが、実態は #valid? のテストですね。

あとインスタンス変数とアクセサメソッドはがっつり混同していました。失礼しました。

Reply:
雑談: 今後、複雑なアプリケーションを作る場合、複数の属性にまたがったバリデーションが出てくることがありますね。
〜〜権限の場合は〜〜が必須、とかそういうやつ。

そういうことも踏まえると、属性単位で定義をするのは幸せになれないですね

Reply:
なるほど。